### ISSUE TITLE
After upgrade to 2.8.10 error "stringification not supported for node type object element"

### ISSUE THREAD

**Issue by eNeRGy164** [arangodb/arangodb#1937]
After upgrading from 2.8.9 to 2.8.10 I get the following exception on my query

> stringification not supported for node type object element (location: arangod/Aql/AstNode.cpp:2352). Please report this error to arangodb.com
(while optimizing plan) (location: arangod/RestHandler/RestCursorHandler.cpp:155). Please report this error to arangodb.com

With 2.8.9 everything worked perfectly.
I'm running on Linux Ubuntu x86_64 (in Azure)

I've simplified my document and query for reproduction
I have a collection "person", with this sample object:
```json
{
  "lastModifiedOn": "2015-04-24T14:20:54Z",
  "name": "Example",
  "sharePointSites": [
    {
      "lastModifiedOn": "2015-04-23T10:22:45Z",
      "name": "Example Site",
      "id": "Site ID",
      "documentLocations": [
        {
          "lastModifiedOn": "2015-04-23T11:20:59Z",
          "name": "First example location",
          "id": "Some ID",
          "documents": []
        },
        {
          "lastModifiedOn": "2015-04-23T11:21:02Z",
          "name": "Second example",
          "id": "Some other ID",
          "documents": []
        }
      ]
    }
  ]
}
```

The following query results in the exception:
```javascript
FOR c IN person COLLECT 
    sharepointsites = c.sharePointSites[* RETURN { id: CURRENT.id, date: CURRENT.lastModifiedOn }],
    documentlocations = c.sharePointSites[*].documentLocations[* RETURN { id: CURRENT.id, date: CURRENT.lastModifiedOn }][**],
    documents = c.sharePointSites[*].documentLocations[*][**].documents[* RETURN { id: CURRENT.id, date: CURRENT.lastModifiedOn }][**]

RETURN { sharepointsites , documentlocations, documents }

```
Removing the inline returns makes the query work, but not returning the data formatted as wanted
```javascript
FOR c IN person COLLECT 
    sharepointsites = c.sharePointSites[*],
    documentlocations = c.sharePointSites[*].documentLocations[*][**],
    documents = c.sharePointSites[*].documentLocations[*][**].documents[*][**]

RETURN { sharepointsites , documentlocations, documents }
```

**Comment by jsteemann** [comment]
I can confirm the issue only appeared in 2.8.10, as a consequence of another bugfix. The bug fixed in 2.8.10 hid some code for the (internal) stringification of `[** RETURN ...]` that became active in 2.8.10 as a consequence of the fix. 
I have fixed that as well, in branches 2.8, 3.0 and devel.

**Comment by pekeler** [comment]
So there's going to be a 2.8.11 soon?

Do you still recommend 2.8.10 for people not using `[** RETURN ...]`?

**Comment by jsteemann** [comment]
What fails in 2.8.10 is the `[* RETURN x]` (or `[** RETURN x]`) with the `x` expression having certain types. For example, if `x` is an object literal, an empty array or a string, it will fail. If `x` is a non-empty array or some non-const expression, it may actually work. That's also the reason why this did not pop up earlier.
So yes, there should be a 2.8.11 soon.

**Comment by jsteemann** [comment]
Fixed in 2.8.11.

**Comment by eNeRGy164** [comment]
Works!
